jetty-ee10/jetty-ee10-servlet/src/test/java/org/eclipse/jetty/ee10/servlet/ErrorPageTest.java

```diff
@@ -390,11 +390,11 @@ public void testBadMessage() throws Exception
         {
             String response = _connector.getResponse("GET /app?baa=%88%A4 HTTP/1.0\r\n\r\n");
             assertThat(response, Matchers.containsString("HTTP/1.1 400 Bad Request"));
-            assertThat(response, Matchers.containsString("ERROR_PAGE: /BadMessageException"));
+            assertThat(response, Matchers.containsString("ERROR_PAGE: /BadMessage"));
             assertThat(response, Matchers.containsString("ERROR_MESSAGE: Unable to parse URI query"));
             assertThat(response, Matchers.containsString("ERROR_CODE: 400"));
-            assertThat(response, Matchers.containsString("ERROR_EXCEPTION: org.eclipse.jetty.http.BadMessageException: 400: Unable to parse URI query"));
-            assertThat(response, Matchers.containsString("ERROR_EXCEPTION_TYPE: class org.eclipse.jetty.http.BadMessageException"));
+            assertThat(response, Matchers.containsString("ERROR_EXCEPTION: org.eclipse.jetty.http.BadMessage$RuntimeException: 400: Unable to parse URI query"));
+            assertThat(response, Matchers.containsString("ERROR_EXCEPTION_TYPE: class org.eclipse.jetty.http.BadMessage$RuntimeException"));
```

Comment:
This bothers me, as inner classes for working with servlet error pages is confusing enough.
Can this please not be an inner class and be it's own top level?

Reply:
The Jetty-12 style is heavily using inner classes of interfaces.   It could be a top level class, but that is against the style we have been using.

Reply:
@joakime see QuietException, HttpFields, HttpCookie, HttpURI, Handler, Content, etc. as examples of this style.

Reply:
@joakime  it is highly unlikely that webapps will map BadMessageException to an error page.   If they want an error page for 400 responses, they just map code 400 rather than a non standard exception.

But see latest changes anyway, as BadMEssageException is back as a top level class.